support-frontend/assets/pages/supporter-plus-landing/setup/setUserStateActions.ts

```diff
@@ -0,0 +1,39 @@
+import type { ThunkDispatch } from 'redux-thunk';
+import { stateProvinceFieldFromString } from 'helpers/internationalisation/country';
+import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
+import type { ContributionsState } from 'helpers/redux/contributionsStore';
+import { defaultUserActionFunctions } from 'helpers/user/defaultUserActionFunctions';
+import type { Action, UserSetStateActions } from 'helpers/user/userActions';
+
+// ----- Actions Creators ----- //
+// This action creator seems to be overridden in order to prevent setting an invalid string as the user's state
+// as we get this from window.guardian - see helpers/user/user.ts#140 on
+// It would make more sense to put this logic in the reducer itself when we refactor to RTK
```

Comment:
Yes, can see most likely overwrites window.guardian.geoip.stateCode setup in user.init. Might be worth moving there at some point in future.